include/boost/gil/extension/io/bmp.hpp

```diff
@@ -7,8 +7,8 @@
 
 /*************************************************************************************************/
 
-#ifndef BOOST_GIL_EXTENSION_IO_BMP_ALL_HPP
-#define BOOST_GIL_EXTENSION_IO_BMP_ALL_HPP
+#ifndef boost_gil_extension_io_bmp_hpp_
```

Comment:
Is this necessary to change the style of include guards to lowercase? (The trailing underscore is not necessary either)

The Boost convention is that all symbols related to preprocessor are underscore-delimited `UPPERCASE` words. That has a purpose and it's a transitive knowledge across Boost codebase.

Reply:
I wasn't aware of that convention, and I have seen plenty of exceptions to that rule. Is it written somewhere, or just folklore ? (The important part is that the full filename be represented to avoid clashes. Doing my review I have seen one or two cases where a wrong guard macro was used, apparently copy&pasted from a different header. Such issues generate funny symptoms...)
So, as I had to adjust all header guards, I just used the convention I was most familiar with. Do you really think I need to change it again ?

Reply:
> Is it written somewhere, or just folklore ? 


Yes, it is here https://www.boost.org/development/header.html : *Boost convention of all uppercase letters, with the header name prefixed by the namespace name, and suffixed with HPP, separated by underscores.*

>  I have seen plenty of exceptions to that rule

grepping through the whole of Boost headers for `ifndef\s+[a-z]+_.+` shows just one in `phoenix/core/detail/index_sequence.hpp` and, interestingly, handful of exceptions in Boost.Python headers, but I suspect it was a recent change :-)

I can re-store upper-case, no problem, just let me know and as soon as you merge, I'll do it.

I understand the role of personal preference, but please let's